source/ingress.go

```diff
@@ -210,6 +234,48 @@ func (sc *ingressSource) filterByAnnotations(ingresses []*networkv1.Ingress) ([]
 	return filteredList, nil
 }
 
+// filterByIngressClass filters a list of ingresses based on a required ingress
+// class
+func (sc *ingressSource) filterByIngressClass(ingresses []*networkv1.Ingress) ([]*networkv1.Ingress, error) {
+	// if no class filter is specified then there's nothing to do
+	if sc.ingressClassNames == nil {
+		return ingresses, nil
+	}
+
+	classNameReq, err := labels.NewRequirement("kubernetes.io/ingress.class", selection.In, sc.ingressClassNames)
```

Comment:
Labels? Is that right? The deprecated `kubernetes.io/ingress.class` annotation is an annotation, not a label.